portal/app/farm_updater/egauge_api.py

```diff
@@ -0,0 +1,128 @@
+
+        return power_values
+
+    # Function to process data from e-gauge and convert to useful power values
+    def processing_egauge_data(self):
+        power_values = dict.fromkeys(self.keys, None)
+        try:
+            resp = requests.get(self.url, auth=HTTPDigestAuth(self.username, self.password))
+            resp.raise_for_status()
+            data_ini = self.get_egauge_data(resp)
+
+        except requests.exceptions.HTTPError as err:
+            print(err)
+            return json.dumps(power_values)
+
+        time.sleep(self.t_sample)
+
+        try:
+            resp = requests.get(self.url, auth=HTTPDigestAuth(self.username, self.password))
+            resp.raise_for_status()
+            data_end = self.get_egauge_data(resp)
+
+        except requests.exceptions.HTTPError as err:
+            print(err)
+            return json.dumps(power_values)
+
+        ts_delta = data_end['ts'] - data_ini['ts']
+        try:
+            int(ts_delta)
+            power_values['ts'] = datetime.datetime.fromtimestamp(int(data_end['ts'])).strftime('%Y-%m-%d %H:%M:%S')
+            power_values['L1 Voltage'] = ((data_end['L1 Voltage'] - data_ini['L1 Voltage']) / ts_delta) / 1000
+            power_values['L2 Voltage'] = ((data_end['L2 Voltage'] - data_ini['L2 Voltage']) / ts_delta) / 1000
+            power_values['Power Circuit 1'] = (data_end['Power Circuit 1'] - data_ini['Power Circuit 1']) / ts_delta
+            power_values['Power Circuit 1*'] = (data_end['Power Circuit 1*'] - data_ini['Power Circuit 1*']) / ts_delta
+            power_values['Power Circuit 2'] = (data_end['Power Circuit 2'] - data_ini['Power Circuit 2']) / ts_delta
+            power_values['Power Circuit 2*'] = (data_end['Power Circuit 2*'] - data_ini['Power Circuit 2*']) / ts_delta
+            power_values['Power Circuit 1 neutral'] = (data_end['Power Circuit 1 neutral'] - data_ini[
+                'Power Circuit 1 neutral']) / ts_delta
+            power_values['Shed Power'] = (data_end['Shed Power'] - data_ini['Shed Power']) / ts_delta
+            power_values['Control Fan Power'] = (data_end['Control Fan Power'] - data_ini[
+                'Control Fan Power']) / ts_delta
+            power_values['Control Fan Power*'] = (data_end['Control Fan Power*'] - data_ini[
+                'Control Fan Power*']) / ts_delta
+
+            json_dict = json.dumps(power_values)
+            return json.dumps(power_values)
+
+        except Exception as e:
+            print('Error retrieving data from E-Gauge API', e)
+            return None
+
+def update_egauge_data():
+    from app.models import FarmDevice
+    try:
+        egauge_data = EgaugeInterface(url=settings.EGAUGE_URL, username=settings.EGAUGE_USER, password=settings.EGAUGE_PASSWORD).processing_egauge_data()
+        print('egauge data: ', egauge_data)
+
+    except Exception as e:
+        print('Error retireving data from egauge api')
+
+
+    if egauge_data is not None:
```

Comment:
move this whole block into the try except above. This control flow is confusing.